optics-th's test suite failed to build with GHC 8.10.2

```
Building test suite 'optics-th-tests' for optics-th-0.3.0.1..
[1 of 3] Compiling Optics.TH.Tests.DuplicateRecordFields ( tests/Optics/TH/Tests/DuplicateRecordFields.hs, dist/build/optics-th-tests/optics-th-tests-tmp/Optics/TH/Tests/DuplicateRecordFields.dyn_o )

tests/Optics/TH/Tests/DuplicateRecordFields.hs:37:34: error:
    • Multiple declarations of ‘foo’
      Declared at: tests/Optics/TH/Tests/DuplicateRecordFields.hs:20:3
                   tests/Optics/TH/Tests/DuplicateRecordFields.hs:37:34
    • In the Template Haskell quotation
        [d| data family F x
            data instance F Int = MkF {foo :: Int} |]
   |
37 |      data instance F Int = MkF { foo :: Int }|])
   |                                  ^^^
```

--- a/optics-th/tests/Optics/TH/Tests/DuplicateRecordFields.hs
+++ b/optics-th/tests/Optics/TH/Tests/DuplicateRecordFields.hs
@@ -32,15 +32,17 @@ foo'' = view foo
 (>::) :: (:::) -> Int -> Int
 (>::) = view (>:)
 
+-- NB we cannot use the field name 'foo' here, because there is already a
+-- definition of 'foo' as a lens in scope (#338).
 $(declareFieldLabels
   [d|data family F x
-     data instance F Int = MkF { foo :: Int }|])
+     data instance F Int = MkF { woo :: Int }|])
 $(declareLenses
   [d|data family G x
      data instance G Int = MkG { bar :: Int }|])
 
 foo''' :: F Int -> Int
-foo''' = view #foo
+foo''' = view #woo
 
 bar' :: G Int -> Int
 bar' = view bar

